src/plugins/drawing-tool/components/drawing-layer.tsx

```diff
@@ -172,6 +172,10 @@ export class DrawingLayerView extends React.Component<DrawingLayerViewProps, Dra
   public handleObjectClick = (e: MouseEvent|React.MouseEvent<any>, obj: DrawingObjectType) => {
     if (!this.props.readOnly && this.currentTool) {
       this.currentTool.handleObjectClick(e, obj);
+      if (this.state.selectedObjects.length> 1 && !(e.shiftKey || e.metaKey)){
+        const filteredArray = this.state.selectedObjects.filter(selObj=> selObj === obj); //correct behavior
+        this.setState(prevState => ({selectedObjects: filteredArray}));
```

Comment:
Could've also written this as 
`        this.setState(prevState => ({selectedObjects: prevState.selectedObjects.filter(selObj => selObj === obj}));`